### ISSUE TITLE
Some rules (e.g., `prefer-to-be`, `prefer-strict-equal`) do not trigger when using `expect.soft()`

### ISSUE THREAD

**Issue by Rafatcb** [vitest-dev/eslint-plugin-vitest#738]
Hello. I've noticed that some ESLint rules do not seem to work when chained with `expect.soft()`.

For example, the `vitest/prefer-to-be` rule correctly warns for a standard `expect().toEqual()` but fails to warn for `expect.soft().toEqual()`.

Code to reproduce:

```js      
it('should trigger warnings', () => {
  expect.soft(1).toEqual(1); // This line incorrectly passes without a warning
  expect(1).toEqual(1); // This line correctly triggers a warning
});
``` 

The linter only reports one warning for the `expect(1).toEqual(1)` line.

```js      
/path/to/example.test.js
  3:13  warning  Use `toBe` instead  vitest/prefer-to-be

✖ 1 problem (0 errors, 1 warning)
```

I would expect two warnings, as both `toEqual` calls should be flagged.

This issue is not limited to `prefer-to-be`. I confirmed that `vitest/prefer-strict-equal` also fails to trigger on `expect.soft(a).toStrictEqual(b)`. 

This suggests the root cause is how the rules parse assertion chains that include modifiers. It likely affects both `.soft()` and the similar [`expect.poll()`](https://vitest.dev/api/expect.html#poll) modifier, though I have not personally tested the latter.

You can see the bug live in [this CodeSandbox](https://codesandbox.io/p/devbox/beautiful-merkle-3tlfp8?file=%2Fexample.test.js%3A3%2C13).

I'm not familiar with ESLint rule development to propose a fix, so I'm reporting this for the team to investigate. Thanks for all your work :)
